harness/determined/searcher/core_search_runner.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from typing import Any, Dict, Optional
+
+import determined as det
+from determined.experimental import client
+from determined.searcher.search_method import SearchMethod
+from determined.searcher.search_runner import SearchRunner
+
+
+class CoreSearchRunner(SearchRunner):
+    def __init__(self, search_method: SearchMethod, context: det.core.Context) -> None:
+        super().__init__(search_method)
+        self.context = context
+        info = det.get_cluster_info()
+        assert info is not None, "CoreSearchRunner only runs on-cluster"
+        self.latest_checkpoint = info.latest_checkpoint
+
+    def run(
+        self,
+        exp_config: Dict[str, Any],
+        context_dir: Optional[str] = None,
+    ) -> None:
+        logging.info("CoreSearchRunner.run")
+
+        if self.latest_checkpoint is not None:
```

Comment:
i slightly prefer moving out load as a separate method and calling it here in run.